cpp/src/arrow/compute/exec/asof_join_node.cc

```diff
@@ -37,13 +39,41 @@
 namespace arrow {
 namespace compute {
 
-// Remove this when multiple keys and/or types is supported
-typedef int32_t KeyType;
+template <typename T, typename V = typename T::value_type>
+inline typename T::const_iterator std_find(const T& container, const V& val) {
+  return std::find(container.begin(), container.end(), val);
+}
+
+template <typename T, typename V = typename T::value_type>
+inline bool std_has(const T& container, const V& val) {
+  return container.end() != std_find(container, val);
+}
+
+typedef uint64_t KeyType;
+typedef uint64_t TimeType;
```

Comment:
Maybe to rename to "OnType" and "ByType" to be consistent with the names that are used in the AsofJoinOptions

Reply:
Actually - not a big deal I think since we refer to the word "time" quite a bit in the implmentation